File: julia/problem_05.jl
# Copyright 2014 Mitchell Kember. Subject to the MIT License.
# Project Euler: Problem 5
# Smallest multiple

module problem_05

function divisible(n)
	for i = 11:19
		if n % i != 0
			return false
		end
	end
	true
end

function main()
	multiple = 20
	while !divisible(multiple)
		multiple += 20
	end
	multiple
end

end # module

File: haskell/Problem05.hs
-- Copyright 2015 Mitchell Kember. Subject to the MIT License.
-- Project Euler: Problem 5
-- Smallest multiple

module Problem05 where

import Common (divides)

import Data.List (find)
import Data.Maybe (fromJust)

divisibleTo19 :: Int -> Bool
divisibleTo19 n = all (flip divides n) [11..19]

solve :: Int
solve = fromJust . find divisibleTo19 $ [20, 40..]
